go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -62,6 +62,13 @@ func NewMaster(kube kube.Interface,
 			}
 			return m.DeleteNode(node)
 		},
+		func(old, new interface{}) error {
+			node, ok := new.(*kapi.Node)
+			if !ok {
+				return fmt.Errorf("object is not a node")
+			}
+			return m.AddNode(node)
```

Comment:
Question: does `AddNode` diff and handle old / stale vs. new data? Usually in the code base our update funcs = delete + add, here it's not the case since we never implemented the update, now that we do though: does it do it right?   

Same question for `n.controller.AddNode` and `n.controller.AddPod`

Reply:
So waaaaay back when before I implemented the workqueues in the HO package, we did do a delete and then an add.

When I wrote the PR to add the workqueues, I was convinced that all adds should be idempotent and made the workqueue work that way. Effectively the update code path became `Add`. That PR merged.

Meanwhile there were no complaints 😄 

Now I've loosed my stance on adds being idempotent (mainly because doing that everywhere is more than I can accomplish solo and I really want to finish this work), the Update code path is back.
As there were no complaints before with Updates just calling `Add`, I decided to keep it that way here.

Eventually, after all the workqueue stuff is implemented and some work has been done to make all adds idempotent, I'd like to revert this PR